core/modules/replica/CreateReplicaJob.cc

```diff
@@ -127,12 +126,13 @@ CreateReplicaJobResult const& CreateReplicaJob::getReplicaData() const {
         "CreateReplicaJob::getReplicaData  the method can't be called while the job hasn't finished");
 }
 
-std::string CreateReplicaJob::extendedPersistentState(SqlGeneratorPtr const& gen) const {
-    return gen->sqlPackValues(id(),
-                              databaseFamily(),
-                              chunk(),
-                              sourceWorker(),
-                              destinationWorker());
+std::map<std::string,std::string> CreateReplicaJob::extendedPersistentState() const {
+    std::map<std::string,std::string> result;
+    result["database_family"]    = databaseFamily();
+    result["timeout_sec"]        = std::to_string(chunk());
+    result["source_worker"]      = sourceWorker();
+    result["destination_worker"] = destinationWorker();
```

Comment:
IIRC `map::insert` is somewhat more efficient than a braces insert. On the order of 4 elements it may not matter much, unless this is in a nested loop or something, but FWIW.

Reply:
...but, are you sure you want to hard code the strings here? are they not duplicated anywhere - it's not a situation where you should define constants?

Reply:
I'm 100% sure I don't need that because each class has its own collection of parameters. And a knowledge of these parameters is supposed to be encapsulated within classes. Values of the parameters stored in a database are supposed to be used in various (yet to be defined) studies. A potential candidate of this will be a Web-based Management & Monitoring Portal (my next project I'm about to work on). If any specific needs in formalizing collections of those parameters will arise then extra steps will be taken. And right now, I'm just trying to follow the conservative approach - do things only as needed :-)

Reply:
And regarding, "...map::insert is somewhat more efficient than a braces insert...", the code has been refactored to return a list of pairs (string,string).